### PR TITLE
Json changes

### PR DESCRIPTION
**Pull request by gombasg**
Add support for parsing and serializing data in JSON format.

### CODE REVIEW

File: panc/src/test/pan/Functionality/function/json1.tpl [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+# @expect="/nlist[@name='profile']/boolean[@name='boolean']='true'"
+# @expect="/nlist[@name='profile']/long[@name='long']='1234'"
+# @expect="/nlist[@name='profile']/double[@name='double']='1234.56'"
+# @expect="/nlist[@name='profile']/string[@name='string']='string'"
+# @expect="/nlist[@name='profile']/list[@name='list']/*[3]='three'"
+# @expect="/nlist[@name='profile']/nlist[@name='hash']/long[@name='b']='2'"
+#
+object template json1;
+
+"/boolean" = json_decode("true");
```
Comment (stdweird) [inline panc/src/test/pan/Functionality/function/json1.tpl]: for sanity reasons, could you also add a check like 
```pan
"/boolean" = json_encode(json_decode("true"));
```
and the inverse `json_decode(json_encode())`

File: panc/src/test/pan/Functionality/function/json1.tpl [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+"/long" = json_decode("1234");
+"/double" = json_decode("1234.56");
+"/string" = json_decode('"string"');
+"/list" = json_decode('[1, 2, "three"]');
```
Comment (stdweird) [inline panc/src/test/pan/Functionality/function/json1.tpl]: i'm a bot surprised this works. i tought all ements in a list have to be the same type in pan.

Comment (stdweird) [inline panc/src/test/pan/Functionality/function/json1.tpl]: nvm i guess. it's probably a consequence of schema validation

**Comment by stdweird** [comment]
@gombasg can you rename the pan test files from `.tpl` to `.pan`? otherwsie, this can be merged

**Comment by jrha** [comment]
@stdweird I've updated @gombasg's branch to rename the `tpl` files, please merge if you are happy with this now.

**Comment by stdweird** [comment]
@jrha thx

**Pull request merged**